src/osx/carbon/font.cpp

```diff
@@ -927,6 +927,11 @@ void wxNativeFontInfo::CreateCTFontDescriptor()
         if ( fontname.empty() )
             fontname = FamilyToFaceName(m_family);
 
+        if ( fontname == "Courier")
```

Comment:
I think we need some explanation here as it doesn't seem to be that obvious, e.g.
```suggestion
        // Courier and Times fonts used to be available everywhere and so are commonly
        // hard-coded in the applications (even though they shouldn't be, and "teletype"
        // or "roman" font family should be used instead), so make sure we still support
        // them even if macOS > 12 doesn't have any fonts with these names any more.
        if ( fontname == "Courier")
```